### ISSUE TITLE
"Address already in use" error randomly occurs when starting Patroni 2.1.0

### ISSUE THREAD

**Issue by EtienneM** [zalando/patroni#2003]
**Describe the bug**

When starting Patroni 2.1.0, I randomly get the following error in my logs:

```
2021-07-09 15:36:18.174856098 +0200 CEST [postgresql-1] 2021-07-09 13:36:18,173 ERROR: Couldn't start a service on '0.0.0.0:8008', please check your `restapi.listen` configuration
2021-07-09 15:36:18.245599227 +0200 CEST [postgresql-1] 2021-07-09 13:36:18,173 ERROR: Failed to reload config_file=/etc/patroni/patroni.yml
2021-07-09 15:36:18.245608423 +0200 CEST [postgresql-1] Traceback (most recent call last):
2021-07-09 15:36:18.245609691 +0200 CEST [postgresql-1] File "/usr/local/lib/python2.7/dist-packages/patroni/__init__.py", line 78, in reload_config
2021-07-09 15:36:18.245610678 +0200 CEST [postgresql-1] self.api.reload_config(self.config['restapi'])
2021-07-09 15:36:18.245612089 +0200 CEST [postgresql-1] File "/usr/local/lib/python2.7/dist-packages/patroni/api.py", line 863, in reload_config
2021-07-09 15:36:18.245613119 +0200 CEST [postgresql-1] self.__initialize(config['listen'], ssl_options)
2021-07-09 15:36:18.245614224 +0200 CEST [postgresql-1] File "/usr/local/lib/python2.7/dist-packages/patroni/api.py", line 768, in __initialize
2021-07-09 15:36:18.245615615 +0200 CEST [postgresql-1] self.__httpserver_init(host, port)
2021-07-09 15:36:18.245665701 +0200 CEST [postgresql-1] File "/usr/local/lib/python2.7/dist-packages/patroni/api.py", line 747, in __httpserver_init
2021-07-09 15:36:18.245669373 +0200 CEST [postgresql-1] HTTPServer.__init__(self, info[0][-1][:2], RestApiHandler)
2021-07-09 15:36:18.245670568 +0200 CEST [postgresql-1] File "/usr/lib/python2.7/SocketServer.py", line 417, in __init__
2021-07-09 15:36:18.245824292 +0200 CEST [postgresql-1] self.server_bind()
2021-07-09 15:36:18.245851871 +0200 CEST [postgresql-1] File "/usr/lib/python2.7/BaseHTTPServer.py", line 108, in server_bind
2021-07-09 15:36:18.245853207 +0200 CEST [postgresql-1] SocketServer.TCPServer.server_bind(self)
2021-07-09 15:36:18.245934030 +0200 CEST [postgresql-1] File "/usr/lib/python2.7/SocketServer.py", line 431, in server_bind
2021-07-09 15:36:18.245936331 +0200 CEST [postgresql-1] self.socket.bind(self.server_address)
2021-07-09 15:36:18.245937316 +0200 CEST [postgresql-1] File "/usr/lib/python2.7/socket.py", line 228, in meth
2021-07-09 15:36:18.245938350 +0200 CEST [postgresql-1] return getattr(self._sock,name)(*args)
2021-07-09 15:36:18.245939610 +0200 CEST [postgresql-1] error: [Errno 98] Address already in use
```

This error occurs randomly but it seems to happen rather often. I fail to reproduce this with Patroni 2.0.2.

**To Reproduce**

Try to start Patroni 2.1.0. You may notice the above-mentioned log message.

**Expected behavior**

No error

**Screenshots**

n/a

**Environment**
- Patroni version: 2.1.0
- PostgreSQL version: 13.3.0
- DCS (and its version): etcd 3.4.16

**Patroni configuration file**

```
etcd:                 
  protocol: https                                                                                                                                                                                                  
  username: REDACTED
  password: REDACTED

bootstrap:   
  dcs:                                                                                                   
    ttl: 60        
    loop_wait: 20  
    retry_timeout: 20 
    maximum_lag_on_failover: 1048576  
    postgresql:
      use_pg_rewind: true
      parameters:
        max_connections: '50'                                                                            
        max_wal_senders: '3'                                                                             
        wal_keep_size: '48MB'                                                                            
        max_locks_per_transaction: '64'                                                                  

  method: pgbackrest            
  pgbackrest:               
    command: /postgresql_bootstrap_replica --username=REDACTED
    keep_existing_recovery_conf: true
  post_bootstrap: /postgresql_post_bootstrap

restapi:                                                                                                 
  certfile: /etc/ssl/databases/server.crt                                                                
  keyfile: /etc/ssl/databases/server.key
  connect_address: 192.168.100.3:8008        

ctl:
  insecure: false
  cacert: /etc/ssl/databases/root.crt
  certfile: /etc/ssl/databases/server.crt

postgresql:
  data_dir: /var/lib/postgresql/13/postgresql0
  bin_dir: /usr/lib/postgresql/13/bin
  config_dir: /etc/postgresql/13/main
  pgpass: /tmp/pgpass0
  recovery_conf:
    restore_command: 'pgbackrest --config=/var/lib/postgresql/13/pgbackrest.conf --stanza=biniou_8547 archive-get %f "%p"'
  parameters:
    unix_socket_directories: '/tmp'
    ssl: 'on'
    ssl_cert_file: '/etc/ssl/databases/server.crt'
    ssl_key_file: '/etc/ssl/databases/server.key'
    ssl_ca_file: '/etc/ssl/databases/root.crt'
    ssl_ciphers: 'HIGH'
    wal_level: 'replica'
    archive_mode: 'on'
    archive_command: 'pgbackrest --config=/var/lib/postgresql/13/pgbackrest.conf --stanza=biniou_8547 archive-push %p'
  create_replica_methods:
    - pgbackrest
    - xxx_basebackup
  pgbackrest:
    command: /usr/bin/pgbackrest --stanza=biniou_8547 --delta restore
    keep_data: true
    no_params: true
  xxx_basebackup:
    command: /postgresql_basebackup.sh
    retries: 2
    verbose: "true"
  pg_hba:
    - hostssl   replication replicator 192.168.100.0/24 md5
    - hostssl   replication replicator 127.0.0.1/32 md5
    - hostnossl replication all        0.0.0.0/0     reject
    - local     postgres    postgresql               peer
    - host all all 0.0.0.0/0 md5
    - host all all ::0/0 md5

tags:
  nofailover: false
  noloadbalance: false
  clonefrom: false
  nosync: false
```

**patronictl show-config**
```
loop_wait: 1800
maximum_lag_on_failover: 1048576
pause: false
postgresql:
  parameters:
    max_connections: '50'
    max_locks_per_transaction: '64'
    max_wal_senders: '3'
    wal_keep_segments: '3'
    wal_keep_size: 48MB
  pg_hba:
  - hostssl replication replicator  md5
  - hostssl replication replicator 127.0.0.1/32 md5
  - hostnossl replication all 0.0.0.0/0 reject
  - local postgres postgresql peer
  - host all all 0.0.0.0/0 md5
  use_pg_rewind: true
retry_timeout: 20
ttl: 2400
```

**Have you checked Patroni logs?**

I provided the log in the "Describe the bug" section

**Have you checked PostgreSQL logs?**

n/a

**Have you tried to use GitHub issue search?**

I didn't found any issue with the error message "Address already in use" in the list of GitHub issues.

**Additional context**

The operating system is Debian Stretch.
The Python version is 2.7.13

**Comment by CyberDem0n** [comment]
Interesting. How did it happen that Patroni gets SIGHUP during the start?

**Comment by CyberDem0n** [comment]
Never mind, it could actually be the change of the /config key in DCS.

**Comment by CyberDem0n** [comment]
@EtienneM, could you please check https://github.com/zalando/patroni/pull/2004

**Comment by EtienneM** [comment]
I attempted to boot ten different clusters with the fix included in #2004 and I didn't faced the issue again. As the problem appeared randomly I can't be 100% sure your fix works, but it seems much better :)
